src/gc.rs

```diff
@@ -41,18 +40,32 @@ pub(crate) enum CollectorState {
     Finalization,
 }
 
+impl CollectorState {
+    fn update(&mut self, new_state: CollectorState) {
+        let err = format!("Invalid GC state transition: {:?} -> {:?}", self, new_state);
```

Comment:
Since only one arm of the `match` references `err`, shouldn't we move it there?

Reply:
Sure. Fixed: 3b68497defa403765875a62aa3bc1c0699bf051e